core/api/src/tasks/schedule/run.ts

```diff
@@ -29,6 +30,10 @@ export class ScheduleRun extends RetryableTask {
     await run.afterBatch();
 
     if (importsCount > 0) {
+      if (process.env.NODE_ENV === "test") {
+        // prevents "job already enqueued" in tests
+        params.count = (params.count || 0) + 1;
+      }
```

Comment:
Which test was running into this error?  There's `specHelper.deleteEnqueuedTasks()` from `actionhero` which might be helpful

Reply:
https://docs.actionherojs.com/modules/spechelper.html#deleteenqueuedtasks

Reply:
the ones that looks this seems to be a risk
```
        // check that the run is enqueued
        const found = await specHelper.findEnqueuedTasks("schedule:run");
        expect(found.length).toEqual(1);
        expect(found[0].args[0].scheduleGuid).toBe(schedule.guid);

        // run the schedule
        const run = await Run.create({
          creatorGuid: schedule.guid,
          creatorType: "schedule",
          state: "running",
        });
        await specHelper.runTask("schedule:run", {
          runGuid: run.guid,
          scheduleGuid: schedule.guid,
        });
```

If it's the same time.
Maybe that's why there are these sleeps in the mysql one
```
      await helper.sleep();
      let foundAgain = await specHelper.findEnqueuedTasks("schedule:run");
      expect(foundAgain.length).toEqual(2);
      await specHelper.runTask("schedule:run", foundAgain[1].args[0]);

      await helper.sleep();
      foundAgain = await specHelper.findEnqueuedTasks("schedule:run");
      expect(foundAgain.length).toEqual(3);
      await specHelper.runTask("schedule:run", foundAgain[1].args[0]);
```

Reply:
I'll remove the change to the task and add sleeps.